nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/main/java/org/nd4j/linalg/jcublas/buffer/BaseCudaDataBuffer.java

```diff
@@ -835,25 +847,32 @@ public void set(long[] data, long length, long srcOffset, long dstOffset) {
      * @param dstOffset
      */
     public void set(float[] data, long length, long srcOffset, long dstOffset) {
+        lazyAllocateHostPointer();
+        allocator.synchronizeHostData(this);
         switch (dataType()) {
             case BOOL: {
                 val pointer = new BytePointer(ArrayUtil.toBytes(data));
                 copyDataFromSrc(pointer,length,offset,dstOffset);
 
             }
             break;
-            case BYTE: {
-                val pointer = new BytePointer(ArrayUtil.toBytes(data));
-                copyDataFromSrc(pointer,length,offset,dstOffset);
 
+            case BFLOAT16: {
+                val pointer = new ShortPointer(ArrayUtil.toBfloats(data));
+                copyDataFromSrc(pointer, length, offset, dstOffset);
             }
             break;
             case UBYTE: {
-                for (int e = 0; e < data.length; e++) {
-                    put(e, data[e]);
-                }
+                data = ArrayUtil.cutBelowZero(data);
+
+            }
```

Comment:
Is this explicitly meant to fall through to the next case? If so, put a note here.